src/test/jtx/Env.h

```diff
@@ -703,6 +704,25 @@ Env::rpc(std::string const& cmd, Args&&... args)
         std::forward<Args>(args)...);
 }
 
+/**
+ * Executes a set of provided functions (Fs...) over a range of versions from
+ * RPC::apiMinimumSupportedVersion to RPC::apiBetaVersion. Each function in
+ * Fs... is expected to take a version number as its argument. This is useful
+ * for running a series of tests or operations that need to be performed on
+ * multiple versions of an API.
+ */
+template <class... Fs>
```

Comment:
You don't need the type anywhere inside the body so you could just use `auto`: 
```cpp
void forAllApiVersions(auto... callables)
```

You could also constraint it to require to be callable with a version number using a concept:
```cpp
template<typename T>
concept VersionedTestCallable = requires(T a, uint32_t v) // use same type as real version for `v`
{
    { a(v) } -> std::same_as<void>;
};
// ...
void forAllApiVersions(VersionedTestCallable auto... fs)
```

Without any of this you still have an ok solution in my opinion.

Reply:
Hi @godexsoft , sorry for the late reply - I've never practically used concepts before so I was learning about it.

```
template <typename T>
concept SingleVersionedTestCallable = requires(T a, uint32_t v)
{
    { a(v) } -> std::same_as<void>;
};

template <typename... T>
concept VersionedTestCallable = (... && SingleVersionedTestCallable<T>);
```

It looks like I'll have to do something like this. Does this look good ?

Reply:
Yea, something like that. I'd think my version also works but i may be wrong - i did not check it. You can experiment on godbolt (compiler explorer) and see what works.